java-checks/src/main/java/org/sonar/java/checks/spring/NullableInjectedFieldsHaveDefaultValueCheck.java

```diff
@@ -0,0 +1,226 @@
+      originalValue.strip().substring(0, currentValue.lastIndexOf('}'))
+      + ":#{null}}"
+      + "\"";
+    String quickFixMessage = "Set null as default value";
+    // Test if the value is defined in a constant that can be fixed as an alternative
+    if (!expression.is(Tree.Kind.STRING_LITERAL)) {
+      quickFixMessage = "Set null as default value locally";
+      computeQuickFixOnOriginalDefinition(expression, replacementValue).ifPresent(quickFixes::add);
+    }
+    // Insert local replacement
+    quickFixes.add(
+      JavaQuickFix.newQuickFix(quickFixMessage)
+        .addTextEdit(JavaTextEdit.replaceTree(expression, replacementValue))
+        .build()
+    );
+    return quickFixes;
+  }
+
+  private static Optional<JavaQuickFix> computeQuickFixOnOriginalDefinition(ExpressionTree expression, String replacementValue) {
+    Symbol symbol;
+    if (expression.is(Tree.Kind.MEMBER_SELECT)) {
+      symbol = ((MemberSelectExpressionTree) expression).identifier().symbol();
+    } else {
+      symbol = ((IdentifierTree) expression).symbol();
+    }
+    Tree declaration = symbol.declaration();
+    if (declaration != null && declaration.is(Tree.Kind.VARIABLE)) {
+      ExpressionTree assignedExpression = ((VariableTree) declaration).initializer();
+      if (assignedExpression != null) {
+        return Optional.of(
+          JavaQuickFix.newQuickFix("Set null as default value")
+            .addTextEdit(JavaTextEdit.replaceTree(assignedExpression, replacementValue))
+            .build());
+      }
+    }
+    return Optional.empty();
+  }
+
+  /**
+   * Maps a variable to an {@link AnnotationsOfInterest} if it is annotated as nullable and injected with a Spring value annotation missing a default value.
+   *
+   * @param variable A field or a parameter
+   * @param valueAnnotationOnParent In the case of a parameter, a possible value annotation without default on the parent method
+   * @return An Optional with a {@link AnnotationsOfInterest} if the member matches. Optional.empty() otherwise.
+   */
+  private static Optional<AnnotationsOfInterest> mapToAnnotationsOfInterest(VariableTree variable, @Nullable AnnotationTree valueAnnotationOnParent) {
+    final AnnotationTree valueAnnotation;
+    if (valueAnnotationOnParent == null) {
+      Optional<AnnotationTree> annotationOnVariable = getValueAnnotationWithoutDefault(variable);
+      if (annotationOnVariable.isEmpty()) {
+        return Optional.empty();
+      }
+      valueAnnotation = annotationOnVariable.get();
+    } else {
+      valueAnnotation = valueAnnotationOnParent;
+    }
+    Optional<AnnotationTree> nullableAnnotation = getNullableAnnotation(variable);
+    return nullableAnnotation.map(annotationTree -> new AnnotationsOfInterest(valueAnnotation, annotationTree));
+  }
+
+  @Nullable
+  private static AnnotationTree extractValueAnnotationOnSetter(MethodTree method) {
+    if (method.simpleName().name().startsWith("set") && method.parameters().size() == 1) {
```

Comment:
As it so happens, @ADarko22 just introduced a helper to check if a method is a setter this morning `MethodTreeUtils#isSetterMethod`: https://github.com/SonarSource/sonar-java/pull/4511/files#diff-96cf5c4851ac66958b366adab42fe06f493aea83a9c7be579c15701839748b16R83